src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -459,14 +467,45 @@ private void CreateNewIndex(Directory dir)
                     //unlock it!
                     IndexWriter.Unlock(dir);
                 }
+
                 //create the writer (this will overwrite old index files)
                 writer = new IndexWriter(dir, FieldAnalyzer, true, IndexWriter.MaxFieldLength.UNLIMITED);
+                if (_directory is ExamineDirectory examineDirectory)
+                {
+                 
+                    if (examineDirectory.GetDeletionPolicy() != null)
+                    {
+                        writer.Dispose();
+                        writer = new IndexWriter(dir, FieldAnalyzer, true, examineDirectory.GetDeletionPolicy(),
+                            IndexWriter.MaxFieldLength.UNLIMITED);
+                    }
 
-                // clear out current scheduler and set the error logging one
-                using (writer.MergeScheduler) { }
-                writer.SetMergeScheduler(new ErrorLoggingConcurrentMergeScheduler(Name,
-                    (s, e) => OnIndexingError(new IndexingErrorEventArgs(this, s, "-1", e))));
+                    if (examineDirectory.GetMergeScheduler() != null)
+                    {
+                        writer.SetMergeScheduler(examineDirectory.GetMergeScheduler());
+                    }
+
+                    if (examineDirectory.IsReadOnly)
+                    {
+                        DocumentWriting += (sender, args) => { args.Cancel = true; };
```

Comment:
Do not bind to our own events, this class in in control of raising it so the logic should be there. This also avoid the problem of not unbinding events.